test/ERC721Drop.t.sol

```diff
@@ -977,6 +977,49 @@ contract ERC721DropTest is Test {
         assertEq(dummyRenderer.someState(), "");
     }
 
+    function test_SupplyRoyaltyPurchase() public setupZoraNFTBase(100) {
```

Comment:
this test is a bit hard to follow - and its one big test testing a bunch of logic - how about breaking it into smaller ones testing the different branches?